test/chaat/db/user_test.clj

```diff
@@ -0,0 +1,71 @@
+(ns chaat.db.user-test
+  (:require [chaat.fixture :as fixture]
+            [clojure.test :refer :all]
+            [chaat.db.user :as db.user]
+            [chaat.model.user :as model.user]
+            [crypto.password.bcrypt :as bcrypt]
+            [java-time.api :as jt]))
+
+(use-fixtures :each fixture/test-fixture)
+
+(defn- time-stub
+  "Return a fixed timestamp"
+  []
+  (jt/instant "2023-06-13T10:07:03.172Z"))
+
+(defn- encrypt-stub
+  "Return a fixed password hash"
+  [_ _]
+  "$2a$11$DoWjFwnL5glpyGqBRgdA3uqoy1glTFVoXP.wesem27g2SL3XFXOHW")
+
+;; no need to test this?
+(deftest new-user?-test
+  (testing ""))
+
+;; no need to test this?
+(deftest user-exists?-test
+  (testing ""))
+
+(deftest insert-test
+  (let [username "john"
+        password "12345678"
+        datasource (:db fixture/test-system)]
+    (testing "Insert new user into user table succeeds"
+      (with-redefs [model.user/get-time-instant time-stub
+                    bcrypt/encrypt encrypt-stub]
+        (let [user-info (model.user/gen-new-user-map username password)
+              actual-result (db.user/insert datasource user-info)
+              actual-result-without-id (update-in actual-result [:result] dissoc :users/id)
+              expected-result {:result
+                               #:users{:username "john",
+                                       :password_hash "$2a$11$DoWjFwnL5glpyGqBRgdA3uqoy1glTFVoXP.wesem27g2SL3XFXOHW",
+                                       :creation_timestamp #inst "2023-06-13T10:07:03.172000000-00:00",
+                                       :display_picture nil},
+                               :error nil}]
+          (is (= true (db.user/user-exists? (datasource) "john")))
+          (is (= expected-result actual-result-without-id))))
```

Comment:
[minor, non-blocking] You might get different opinions on this, but wherever reasonable, I usually prefer inlining the expected and actual result over here, rather than hiding them behind a binding.

```suggestion
         (is (= {:result
                 #:users {:username "john",
                          :password_hash "$2a$11$DoWjFwnL5glpyGqBRgdA3uqoy1glTFVoXP.wesem27g2SL3XFXOHW",
                          :creation_timestamp #inst "2023-06-13T10:07:03.172000000-00:00",
                          :display_picture nil}
                 :errors nil}
                (-> (db.user/insert datasource user-info)
                    (update-in [:result] dissoc :id))))
```

From WEWUT (in our digital library), under the section titled ("Expect Literals")
> Tests that use literal expected values are easier to read; when an expected value is simply a literal you’re able to focus almost exclusively on the actual value.

I tend to agree.

Reply:
I was confused about this. Because we were dealing with multi-line maps here, I felt using a binding was cleaner. 
I see the advantage in inlining, and will tend to that in the future. I would however like to discuss if there are situations when bindings would be better when we next speak.

Reply:
I'll start by writing new tests using the inline approach, and eventually come back to refactor these.